config/config.go

```diff
@@ -202,3 +257,26 @@ func GetDefaultCluster(requestIDHeader, activityName, cmdLine, cmdAPIEndpoint st
 	}
 	return "", errors.New(orgsResponse.StatusText)
 }
+
+// migrateConfigDir migrates a configuration directory from the old
+// default path to the new default path. Conditions:
+// - old config dir exists
+// - new config dir does not exist
+func migrateConfigDir() error {
+	_, err := os.Stat(DefaultConfigDirPath)
+	if !os.IsNotExist(err) {
+		// new config dir already exists
+		return nil
+	}
+
+	_, err = os.Stat(LegacyConfigDirPath)
+	if os.IsNotExist(err) {
+		// old config dir does not exist
+		return nil
+	}
+
+	// ensure ~/.config exists
+	os.MkdirAll(path.Dir(DefaultConfigDirPath), 0700)
```

Comment:
Bit of a nitpick but os.MkdirAll can return an error. Should be OK as its the home dir but might be good to check for it.